src-java/stats-topology/stats-storm-topology/src/test/java/org/openkilda/wfm/topology/stats/StatsTopologyTest.java

```diff
@@ -426,6 +494,519 @@ public void flowStatsTest() {
+    }
+
+    /**
+     * 3 switch ha-flow flow-stats test.
+     * ''''''/----3
+     * 1----2
+     */
+    @Test
+    public void haFlowStatsTest3() {
+        HaFlow haFlow = createHaFlow3();
+
+        sendUpdateHaFlowPathInfo(haFlow);
+        //forward
+        FlowStatsEntry forwardIngress = new FlowStatsEntry(1, haFlow.getForwardPath().getCookie().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_1, Collections.singletonList(forwardIngress)));
+        validateHaFlowStats(forwardIngress, haFlow.getForwardPath().getCookie(), SWITCH_ID_1, true,
+                false, false, SUB_FLOW_ID_SHARED);
+
+        FlowStatsEntry forwardYPoint = new FlowStatsEntry(1, haFlow.getForwardPath().getCookie().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_2, Collections.singletonList(forwardYPoint)));
+        validateHaFlowStats(forwardYPoint, haFlow.getForwardPath().getCookie(), SWITCH_ID_2, false,
+                true, true, SUB_FLOW_ID_SHARED);
+
+        FlowStatsEntry forwardEgressPoint1 = new FlowStatsEntry(1, COOKIE_FORWARD.toBuilder()
+                .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build().getValue(), 150L,
+                300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_3, Collections.singletonList(forwardEgressPoint1)));
+        validateHaFlowStats(forwardEgressPoint1, haFlow.getForwardPath().getCookie().toBuilder()
+                        .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build(), SWITCH_ID_3, false,
+                true, false, SUB_FLOW_ID_1);
+
+        FlowStatsEntry forwardEgressPoint2 = new FlowStatsEntry(1, haFlow.getForwardPath().getCookie().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_2, Collections.singletonList(forwardEgressPoint2)));
+        validateHaFlowStats(forwardEgressPoint2, haFlow.getForwardPath().getCookie(), SWITCH_ID_2, false,
+                true, false, SUB_FLOW_ID_SHARED);
+
+        //reverse:
+        FlowSegmentCookie cookieReverse = haFlow.getReversePath().getCookie();
+
+        FlowStatsEntry reverseIngressPoint1 = new FlowStatsEntry(1, cookieReverse.toBuilder()
+                .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build().getValue(), 150L,
+                300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_3, Collections.singletonList(reverseIngressPoint1)));
+        validateHaFlowStats(reverseIngressPoint1, cookieReverse.toBuilder()
+                        .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build(), SWITCH_ID_3, true,
+                false, false, SUB_FLOW_ID_1);
+
+        FlowStatsEntry reverseYPoint1 = new FlowStatsEntry(1, cookieReverse.toBuilder()
+                .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build().getValue(), 150L,
+                300L, 10, 10);
```

Comment:
yo use 150, 300, 10, 10 stats everywhere. Please you different values, because we must miss a bug if all will be equal